src/common/snippets/src/utils.cpp

```diff
@@ -87,16 +87,50 @@ ov::PartialShape get_reordered_planar_shape(const ov::PartialShape& shape, const
     return reordered_shape;
 }
 
+IShapeInferSnippets::VectorDims partial_shape_to_vector_dims(const PartialShape& pshape) {
+    IShapeInferSnippets::VectorDims result;
+    result.reserve(pshape.size());
+    for (const auto& d : pshape)
+        result.push_back(d.is_dynamic() ? IShapeInferSnippets::DYNAMIC_DIMENSION : d.get_length());
+    return result;
+}
+
+ov::PartialShape vector_dims_to_partial_shape(const IShapeInferSnippets::VectorDims& vdims) {
+    ov::PartialShape result;
+    result.reserve(vdims.size());
+    for (const auto& v : vdims)
+        result.push_back(v != IShapeInferSnippets::DYNAMIC_DIMENSION ?
+                         Dimension(static_cast<Dimension::value_type>(v)) :
+                         Dimension());
+    return result;
+}
+
 ov::PartialShape get_port_planar_shape(const Input<Node>& in) {
-    const auto& port = lowered::PortDescriptorUtils::get_port_descriptor_ptr(in);
+    const auto& port = snippets::lowered::PortDescriptorUtils::get_port_descriptor_ptr(in);
     return utils::get_reordered_planar_shape(ov::Shape{port->get_shape()}, port->get_layout());
 }
 
 ov::PartialShape get_port_planar_shape(const Output<Node>& out) {
-    const auto& port = lowered::PortDescriptorUtils::get_port_descriptor_ptr(out);
+    const auto& port = snippets::lowered::PortDescriptorUtils::get_port_descriptor_ptr(out);
     return utils::get_reordered_planar_shape(ov::Shape{port->get_shape()}, port->get_layout());
 }
 
+namespace lowered {
+
+std::vector<size_t> get_planar_shape(const std::vector<size_t>& shape, const std::vector<size_t>& layout) {
```

Comment:
We already the same function for `PartialShape`: `get_reordered_planar_shape(const ov::PartialShape& shape, const std::vector<size_t>& layout)`. As I understand, you created new for `VectorDims` (btw, I think we need to rename vector to `VectorDims` to avoid overheads on conversion between `PartialShape` and `VectorDims`. I agree. But what do you think about to have the similar function names at least? 😄 Moreover I'm not sure that this function should be in namespace `lowered` as well because we work with ordinary instances

Reply:
It's in `lowered` because we need it primary for LinearIR and expressions that are in `lowered` as well. Now I'm thinking, maybe we should move all shape_infer there as well because it also works only on LIR 🤔. What do you think?

Reply:
> (btw, I think we need to rename vector to `VectorDims` to avoid overheads on conversion between `PartialShape` and `VectorDims`

Didn't get it, could you elaborate please?

Reply:
> Didn't get it, could you elaborate please?

I'm sorry I missed closing bracket:

> As I understand, you created new for `VectorDims` (btw, I think we need to rename vector to `VectorDims`**)** to avoid overheads on conversion between `PartialShape` and `VectorDims`

This comment is actual still: Probably we should rename `std::vector<size_t>` to `VectorDims`


>Now I'm thinking, maybe we should move all shape_infer there as well because it also works only on LIR 🤔. What do you think?

BTW it's cool idea. We make shape inference on lowered level but its'not in this namespace  - it's strange a little bit. Could you please if it's possible to move all shape inference things on lowered level to `lowered` namespace?

Reply:
As discussed offline, renamed shape-manipulation utility function and removed `utility::lowered` namespace
Also moved shape inference from `snippets/lowered` to `snippets`.
Please take a look